### PR TITLE
Fix build in tests/overview when groupid is defined

### PR DESCRIPTION
**Pull request by mimi1vx**
Fixes https://progress.opensuse.org/issues/98258

### CODE REVIEW

**Review by okurz — CHANGES_REQUESTED** [review]
s/wheen/when/

**Review by okurz — CHANGES_REQUESTED** [review]
Please add tests for the issue. I recommend to use https://en.wikipedia.org/wiki/Test-driven_development although for the issue at hand it's obviously too late already :D

File: lib/OpenQA/Schema/ResultSet/Jobs.pm [hunk 1]
```diff
@@ -317,9 +317,12 @@ sub prepare_complex_query_search_args ($self, $args) {
             push(@conds, {'me.id' => {-in => $subquery->get_column('job_id')->as_query}});
         }
 
-        for my $key (qw(build distri version flavor arch test machine)) {
+        for my $key (qw(distri version flavor arch test machine)) {
             push(@conds, {"me." . uc($key) => $args->{$key}}) if $args->{$key};
         }
+        if (defined $args->{build}) {
+            push(@conds, {"me.BUILD" => {-in => $args->{build}}});
```
Comment (Martchus) [inline lib/OpenQA/Schema/ResultSet/Jobs.pm:324]: Ok, so apparently `$args->{build}` is an array ref here. I hope that's always¹ the case and that the values contained by that array ref are sufficiently validated.

¹ This function might be called in multiple places.

Comment (mimi1vx) [reply lib/OpenQA/Schema/ResultSet/Jobs.pm:324]: ARRAYREF or Scalar .. it comes from `_compose_job_overview_search_args` if isn't defined in query or from query ..

Comment (Martchus) [reply lib/OpenQA/Schema/ResultSet/Jobs.pm:324]: But we call `complex_query` (which calls this function) also in other places, e.g. `OpenQA::WebAPI::Controller::API::V1::Job::list`. Have you checked the other occurrences as well?

Comment (mimi1vx) [reply lib/OpenQA/Schema/ResultSet/Jobs.pm:324]: hmm, you're right

Comment (mimi1vx) [reply lib/OpenQA/Schema/ResultSet/Jobs.pm]: @Martchus probadly overkill but in this case better safe than sorry 🗡️

File: lib/OpenQA/Schema/ResultSet/Jobs.pm [hunk 2]
```diff
@@ -317,9 +317,16 @@ sub prepare_complex_query_search_args ($self, $args) {
...
+        for my $key (qw(distri version flavor arch test machine)) {
             push(@conds, {"me." . uc($key) => $args->{$key}}) if $args->{$key};
         }
+        if (defined $args->{build}) {
+            my $build = $args->{build};
+            if (ref $build eq 'ARRAY') {
+                push(@conds, {"me.BUILD" => {-in => $build}});
+            }
+            else { push(@conds, {"me.BUILD" => $build}) }
+        }
```
Comment (okurz) [inline lib/OpenQA/Schema/ResultSet/Jobs.pm:329]:
```suggestion
        if (my $build = $args->{build}) {
                push @conds, ref $build eq 'ARRAY' ? {"me.BUILD" => {-in => $build}} : {"me.BUILD" => $build};
        }
```

Comment (okurz) [reply lib/OpenQA/Schema/ResultSet/Jobs.pm]: I *love* the ternary operator and it's something I look for in all programming languages I use :)

**Review by okurz — CHANGES_REQUESTED** [review]
in your second git commit s/overwiew/overview/ but likely you can just squash the git commit into the first one. Test is part of fix.

File: lib/OpenQA/Schema/ResultSet/Jobs.pm [hunk 3]
```diff
@@ -317,9 +317,12 @@ sub prepare_complex_query_search_args ($self, $args) {
             push(@conds, {'me.id' => {-in => $subquery->get_column('job_id')->as_query}});
         }
 
-        for my $key (qw(build distri version flavor arch test machine)) {
+        for my $key (qw(distri version flavor arch test machine)) {
             push(@conds, {"me." . uc($key) => $args->{$key}}) if $args->{$key};
         }
+        if (my $build = $args->{build}) {
+            push @conds, ref $build eq 'ARRAY' ? {"me.BUILD" => {-in => $build}} : {"me.BUILD" => $build};
```
Comment (Martchus) [inline lib/OpenQA/Schema/ResultSet/Jobs.pm:324]: I would not duplicate the common part and we generally use single quotes when not interpolation is required (despite not all old code has been converted to that style):
```suggestion
            push @conds, {'me.BUILD' => (ref $build eq 'ARRAY' ? {-in => $build} : $build)};
```

Comment (okurz) [reply lib/OpenQA/Schema/ResultSet/Jobs.pm:324]: my suggestion missed that part, sorry

Comment (okurz) [reply lib/OpenQA/Schema/ResultSet/Jobs.pm]: Actually not fixed:

```patch
-    push @conds, ref $build eq 'ARRAY' ? {'me.BUILD' => {-in => $build}} : {'me.BUILD' => $build};
+   push @conds, {'me.BUILD' => ref $build eq 'ARRAY' ? {-in => $build} : $build};

**Review by okurz — APPROVED** [review]

**Review by Martchus — APPROVED** [review]

**Pull request merged**